File: LabTab-Back/src/main/java/cl/labtab/api/dtos/request/CreatePaymentRequest.java
package cl.labtab.api.dtos.request;

import cl.labtab.api.common.enums.PaymentMethodEnum;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.PositiveOrZero;

import java.math.BigDecimal;
import java.util.UUID;

public record CreatePaymentRequest(
        @NotNull UUID billId,
        @NotNull @Positive BigDecimal amount,
        @PositiveOrZero BigDecimal tipAmount,
        @NotNull @Positive BigDecimal totalAmount,
        @NotNull PaymentMethodEnum method,
        String provider,
        String externalTransactionId,
        String currency,
        UUID dineGuestId
) {
}

File: labtab_app_v1/lib/features/payment/data/models/create_payment_request.dart
class CreatePaymentRequest {
  final String billId;
  final int amount;
  final int tipAmount;
  final int totalAmount;
  final String method;
  final String? provider;
  final String? externalTransactionId;
  final String currency;
  final String? dineGuestId;

  const CreatePaymentRequest({
    required this.billId,
    required this.amount,
    this.tipAmount = 0,
    required this.totalAmount,
    required this.method,
    this.provider,
    this.externalTransactionId,
    this.currency = 'CLP',
    this.dineGuestId,
  });

  Map<String, dynamic> toJson() => {
        'billId': billId,
        'amount': amount,
        'tipAmount': tipAmount,
        'totalAmount': totalAmount,
        'method': method,
        if (provider != null) 'provider': provider,
        if (externalTransactionId != null)
          'externalTransactionId': externalTransactionId,
        'currency': currency,
        if (dineGuestId != null) 'dineGuestId': dineGuestId,
      };
}
